src/autotester/server/utils/tests/file_management_test.py

```diff
@@ -0,0 +1,131 @@
+from autotester.server.utils.file_management import (
+    clean_dir_name,
+    random_tmpfile_name,
+    recursive_iglob,
+    copy_tree,
+    copy_test_script_files,
+    ignore_missing_dir_error,
+    move_tree,
+    fd_open,
+    fd_lock,
+)
+from autotester.server.utils import redis_management
+import os.path
+import re
+import tempfile
+import shutil
+from autotester.config import config
+import subprocess
+
+FILES_DIRNAME = config["_workspace_contents", "_files_dir"]
+
+
+def test_clean_dir_name():
+    a = "markus/address"
+    b = a.replace("/", "_")
+    assert clean_dir_name(a) == b
+
+
+def test_random_tmpfile_name():
+    tmp_file = random_tmpfile_name()
+    dir_name = os.path.dirname(tmp_file)
+    base_name = os.path.basename(tmp_file)
+    assert os.path.exists(dir_name) and re.match(
+        r"[+\-]" "0" "[xX]" "0|" "([1-9A-Fa-f][1-9A-Fa-f]*)", base_name
+    )
+
+
+def test_recursive_iglob():
+    root_dir = tempfile.mkdtemp()
+    fd, file = tempfile.mkstemp(dir=root_dir)
+    sub_dir = tempfile.mkdtemp(dir=root_dir)
+    file_or_dir = list(recursive_iglob(root_dir))
+    dir = []
+    files = []
+    for i in file_or_dir:
+        dir.append(i[1]) if i[0] == "d" else files.append(i[1])
+    assert all([os.path.exists(d) for d in dir])
+    assert all([os.path.exists(f) for f in files])
+    assert sub_dir == dir[0]
+    assert file == files[0]
+    shutil.rmtree(root_dir)
+    os.close(fd)
+
+
+def test_copy_tree():
```

Comment:
Same comment as above. There should be multiple tests for multiple tree structures. 

I would recommend thinking about writing the tests in a way that you can reuse a lot of the code for all of these tests.

I would recommend reading up on and using [pytest fixtures](https://docs.pytest.org/en/latest/fixture.html) as much as possible to reduce the amount of code you need to write